python/hail/methods/family_methods.py

```diff
@@ -188,24 +185,109 @@ def mendel_errors(dataset, pedigree) -> Tuple[Table, Table, Table, Table]:
     |   12 | HomRef  | Any     | HomVar | HemiY      | Dad, Kid      |
     +------+---------+---------+--------+------------+---------------+
 
+    See Also
+    --------
+    :func:`.mendel_error_code`
+
     Parameters
     ----------
     dataset : :class:`.MatrixTable`
-        Dataset.
     pedigree : :class:`.Pedigree`
-        Sample pedigree.
 
     Returns
     -------
     (:class:`.Table`, :class:`.Table`, :class:`.Table`, :class:`.Table`)
-        Four tables as detailed in notes with Mendel error statistics.
     """
-
-    dataset = require_biallelic(dataset, 'mendel_errors')
-
-    kts = dataset._jvds.mendelErrors(pedigree._jrep)
-    return Table(kts._1()), Table(kts._2()), \
-           Table(kts._3()), Table(kts._4())
+    source = call._indices.source
+    if not isinstance(source, MatrixTable):
+        raise ValueError("'mendel_errors': expected 'call' to be an expression of 'MatrixTable', found {}".format(
+            "expression of '{}'".format(source.__class__) if source is not None else 'scalar expression'))
+
+    source = source.select_entries(__GT=call)
+    dataset = require_biallelic(source, 'mendel_errors')
+    tm = trio_matrix(dataset, pedigree, complete_trios=True)
+    tm = tm.select_entries(mendel_code=hl.mendel_error_code(
```

Comment:
In the Scala code, any proband with missing sex was filtered out. Do we still need this?

Reply:
I don't really think so. The autosomes should definitely work either way.